### ISSUE TITLE
Simplify the current, breaking, Keyd module & configs

### ISSUE THREAD

**Issue by carlosvigil** [NixOS/nixpkgs#284797]
### Describe the bug
Dictionary/Object based config of Keyd has resulted in many (closed) issues and overcomplicated solutions. The original config structure is an INI styled `\n` seperated file. The current implementation is exceedingly complicated for what should be a `\n` seperated string or array/list. 

It also blocks the use of layout files. Keymap layouts can be `include`d in `.conf` files. The `keyd.<keyboards>` object creates `.conf` files with a forced `keyd.<keyboards>.ids` (resulting in : `[ids]\n"*"`) field, both results being explicitly forbidden by Keyd in their manpage under layouts.

### Steps To Reproduce
1. Install the latest Keyd from stable or unstable (2.4.2/2.4.3)
2. Port a keyd config to the nix options for the keyd package

### Expected behavior
The clear, concise configurability of Keyd as intended by its design. Produce layout files with no file extention, and without a forced `[ids]` field. Allow the use of a simple string to write `.conf` files. Allow access to the Keyd binary.

### Additional context
Maintainers should experience how this exacerbates the UX, by porting original keyd configs to the current form of various dictionaries: which don't allow certain symbols on either <key> or <value>, all keys and values possibly requiring individual wrapping in quotations, and then semicolon termination on each, then, subsequent wraps of curly braces. All to find out it doesn't work, and also without access to the keyd binary for easier debugging. Then, compare that experience to building and configuring Keyd on any other linux distribution in 5-15 minutes. Perhaps that will deliver the undertanding that dictionaries are an exceedingly nonoptimal solution to a `\n` terminated string.

### Notify maintainers

@woojiq @peterhoeg 

### Metadata
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.74, NixOS, 24.05 (Uakari), 24.05pre576254.ae5c332cbb58`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by woojiq** [comment]
I agree with you and feel a little guilty about what I created :) This was my first contribution to nixpkgs and I didn't know the best practices. I didn't think that maybe the best solution would be not to introduce complex configuration and just write files as they are.

I'm a bit busy right now. So I'd appreciate it if someone could do it, it's actually going to be a relatively easy change. And I don't really want to do it myself, because I have the feeling that I've already tired people with my breaking changes 🤓

You should ask nixpkgs maintainers about this situation before changing, because all configs are now migrated to `.settings` (instead of `extraConfig`) and writing configs as strings is not desirable afaik.

**Comment by peterhoeg** [comment]
One more thing - we should probably make the socket dependent on the actual version (ref the comments) and stick it into `passthru' so we can reference it in the module rather than having it hardcoded twice.
